crates/sats/src/time_duration.rs

```diff
@@ -69,6 +70,16 @@ impl TimeDuration {
                 .expect("Duration overflows i64 microseconds"),
         )
     }
+
+    /// Returns `Some(self + other)`, or `None` if that value would be out of bounds for [`TimeDuration`].
+    pub fn checked_add(self, other: Self) -> Option<Self> {
+        self.to_micros().checked_add(other.to_micros()).map(Self::from_micros)
+    }
+
+    /// Returns `Some(self - other)`, or `None` if that value would be out of bounds for [`TimeDuration`].
+    pub fn checked_sub(self, other: Self) -> Option<Self> {
+        self.to_micros().checked_sub(other.to_micros()).map(Self::from_micros)
+    }
 }
```

Comment:
I was wondering about also adding `from_seconds` and `from_milliseconds` taking i32. Not having those makes blackholio messy. If they are added, we should add them to the C# SDK as well.